static/app/views/organizationStats/getReasonGroupName.ts

```diff
@@ -153,12 +163,17 @@ function getInvalidReasonGroupName(reason: string): string {
       return group;
     }
   }
-  // 2. If there is no direct match check if there is a match for the baseReason
-  const baseReason = reason.split(':')[0];
-  for (const [group, reasons] of Object.entries(invalidReasonsGroup)) {
-    if (reasons.includes(baseReason as DiscardReason)) {
-      return group;
+  // 2. If there is no direct match check if there is a match for any of the baseReasons
+  // starting with the longest.
+  let lastColonIndex = reason.lastIndexOf(':');
+  while (lastColonIndex !== -1) {
+    const baseReason = reason.substring(0, lastColonIndex);
+    for (const [group, reasons] of Object.entries(invalidReasonsGroup)) {
+      if (reasons.includes(baseReason as DiscardReason)) {
+        return group;
+      }
     }
+    lastColonIndex = baseReason.lastIndexOf(':');
```

Comment:
you could use a more functional approach:

```
const parts = reason.split(':');
  const baseReasons = parts
    .slice(0, -1)
    .map((_, i) => parts.slice(0, parts.length - 1 - i).join(':'));

  for (const baseReason of baseReasons) {
    for (const [group, reasons] of Object.entries(invalidReasonsGroup)) {
      if (reasons.includes(baseReason as DiscardReason)) {
        return group;
      }
    }
  }
```

Reply:
Good point, that is indeed nicer, I updated it 🙏🏻